src/fio/fioEngine.js

```diff
@@ -287,192 +264,220 @@ export class FioEngine extends CurrencyEngine {
     } = action.action_trace
     let nativeAmount
     let actorSender
-    let name
-    let memo = ''
+    let name = ''
+    let notes = ''
     let networkFee = '0'
-    let currencyCode = 'FIO'
+    let otherParams: {
+      isTransferProcessed?: boolean,
+      isFeeProcessed?: boolean
+    } = {}
+    const currencyCode = this.currencyInfo.currencyCode
     const ourReceiveAddresses = []
     if (action.block_num <= this.walletLocalData.otherData.highestTxHeight) {
       return action.block_num
     }
     if (trxName !== 'trnsfiopubky' && trxName !== 'transfer') {
       return action.block_num
     }
+
+    // Transfer funds transaction
     if (trxName === 'trnsfiopubky') {
       nativeAmount = data.amount.toString()
       actorSender = data.actor
-      memo = `Recipient Address: ${data.payee_public_key}`
       if (data.payee_public_key === this.walletInfo.keys.publicKey) {
-        name = actorSender
         ourReceiveAddresses.push(this.walletInfo.keys.publicKey)
         if (actorSender === actor) {
           nativeAmount = '0'
         }
       } else {
-        name = data.payee_public_key
         nativeAmount = `-${nativeAmount}`
       }
 
+      const index = this.findTransaction(
+        currencyCode,
+        action.action_trace.trx_id
+      )
+      // Check if fee transaction have already added
+      if (index > -1) {
+        const existingTrx = this.transactionList[currencyCode][index]
+        otherParams = { ...existingTrx.otherParams }
+        if (
```

Comment:
check if `!otherParams.isTransferProcessed && !otherParams.isFeeProcessed`, console the error and continue